src/openrct2-ui/UiContext.Linux.cpp

```diff
@@ -113,16 +115,72 @@ namespace OpenRCT2::Ui
             }
         }
 
+        int ShowMenuDialog(const std::vector<std::string>& options, const std::string& title, const std::string& text) override
+        {
+            std::string executablePath;
+            DIALOG_TYPE dtype = GetDialogApp(&executablePath);
+
+            size_t longest_string = 0;
+            for(const auto& option : options) {
+                if(option.size() > longest_string) {
+                    longest_string = option.size();
+                }
+            }
+
+            // If zenity and kdialog were sane, they would support some kind of
+            // automatic scaling, however they do not, otherwise these windows will be
+            // ridiculously tiny. This is an approximation 
+            int width = (longest_string + 1) * 8;
+            int height = (options.size() + 1) * 8;
+
+            switch(dtype)
+            {
+                case DIALOG_TYPE::ZENITY:
+                {
+                    auto sb = StringBuilder();
+                    sb.Append(reinterpret_cast<utf8*>(String::Format("zenity --list --column '' --width=%d --height=%d", width, height)));
+                    for (const auto& option : options) {
+                        sb.Append(reinterpret_cast<utf8*>(String::Format(" '%s'", option.c_str())));
+                    }
+                    sb.Append(reinterpret_cast<utf8*>(String::Format(" --title '%s' --text '%s'", title.c_str(), text.c_str())));
+
+                    std::string buff;
+                    Platform::Execute(sb.GetBuffer(), &buff);
+                    return std::find(options.begin(), options.end(), buff) - options.begin();
+                }
+                case DIALOG_TYPE::KDIALOG:
+                {
+                    auto sb = StringBuilder();
+                    sb.Append(reinterpret_cast<utf8*>(String::Format("kdialog --geometry %dx%d --title '%s' --menu ",
+                        width, height, title.c_str())));
+                    sb.Append(reinterpret_cast<utf8*>(String::Format(" '%s'", text.c_str())));
+                    for (const auto& option : options) {
```

Comment:
This does not adhere to our code style. Make sure to use `clang-format` on every commit. If set up correctly, you can even do `git clang-format`.